### PR TITLE
Ocd 3292 - Add Specific d2, d3, d10 Activity to Questionable Activity Report

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-3292

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/concept/QuestionableActivityTriggerConcept.java [hunk 1]
```diff
@@ -33,6 +33,15 @@
...
             "ICS added when Certification Criteria 170.315(b)(3) is on a current listing"
     ),
+    NON_CURES_CRITERIA_ADDED_TO_NEW_LISTING(
+            "Old version of Certification Criteria added for new listing"
+    ),
+    NON_CURES_CRITERIA_ADDED_TO_EXISTING_LISTING(
+            "Old version of Certification Criteria added for existing listing"
+    ),
+    NON_CURES_CRITERIA_AND_ICS_ADDED_TO_EXISTING_LISTING(
+            "Old version of Certification Criteria changed to ICS"
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/concept/QuestionableActivityTriggerConcept.java]: Same comment as in the data-model PR.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java [hunk 2]
```diff
@@ -663,6 +569,119 @@ private boolean hasICS(CertifiedProductSearchDetails listing) {
         }
     }
 
+    public QuestionableActivityListingDTO checkNonCuresAuditCriteriaOnCreate(CertifiedProductSearchDetails newListing) {
+        QuestionableActivityListingDTO activity = null;
+        if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
+            // If ICS=0 and they attested to D2, D3, or D10
+            if (!newListing.getIcs().getInherits()) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: is there any chance newListing.getIcs() could be null?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java [hunk 3]
```diff
@@ -663,6 +569,119 @@ private boolean hasICS(CertifiedProductSearchDetails listing) {
...
+            }
+        }
+        return activity;
+    }
+
+    public QuestionableActivityListingDTO checkNonCuresAuditCriteriaAndAddedIcsOnEdit(CertifiedProductSearchDetails origListing,
+            CertifiedProductSearchDetails newListing) {
+        QuestionableActivityListingDTO activity = null;
+        if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
+            if (!origListing.getIcs().getInherits() && newListing.getIcs().getInherits()) { // ICS has been added
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: Could getIcs() be null? I am thinking specifically if this code gets called when editing a legacy 2014 listing? I'm not sure it does get called but just bringing it up.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java [hunk 4]
```diff
@@ -663,6 +569,119 @@ private boolean hasICS(CertifiedProductSearchDetails listing) {
...
+                activity.setAfter(criteriaNumbers);
+            }
+        }
+        return activity;
+    }
+
+    public QuestionableActivityListingDTO checkNonCuresAuditCriteriaAndAddedIcsOnEdit(CertifiedProductSearchDetails origListing,
+            CertifiedProductSearchDetails newListing) {
+        QuestionableActivityListingDTO activity = null;
+        if (ff4j.check(FeatureList.EFFECTIVE_RULE_DATE)) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: The ticket says if the listing's certification date is on or after rule effective date but that doesn't seem like that is what this is checking.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/concept/QuestionableActivityTriggerConcept.java (hunk 1 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/concept/QuestionableActivityTriggerConcept.java]: This is the language from ONC.  Jim got it for me.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java (hunk 4 above)
Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: Addressed.

**Pull request merged**